subsys/bluetooth/controller/hci_driver.c

```diff
@@ -68,6 +68,30 @@ BUILD_ASSERT(!IS_ENABLED(CONFIG_BT_CENTRAL) ||
 BUILD_ASSERT(!IS_ENABLED(CONFIG_BT_PERIPHERAL) ||
 			 (CONFIG_BT_CTLR_SDC_PERIPHERAL_COUNT > 0));
 
+#if defined(CONFIG_BT_HCI_ACL_FLOW_CONTROL)
+/*
+ * The Host will generate up to acl_pkts number of Host Number of Completed Packets command plus a
+ * number of normal HCI commands, as such we need to ensure the tx command buffer count is big
+ * enough to not block incoming ACKs from the host.
+ *
+ * When Controller to Host data flow control is supported, ensure that BT_BUF_CMD_TX_COUNT is
+ * greater than or equal to (BT_BUF_ACL_RX_COUNT + Ncmd), where Ncmd is the supported maximum
+ * Num_HCI_Command_Packets.
+ *
+ * The SDC controller (currently) does not support Num_HCI_Command_Packets > 1, which means Ncmd
+ * is always 1.
+ */
+#if defined(CONFIG_BT_CONN) && defined(CONFIG_BT_HCI_HOST)
+BUILD_ASSERT(BT_BUF_ACL_RX_COUNT < CONFIG_BT_BUF_CMD_TX_COUNT,
+	     "Too low HCI command buffers compared to ACL Rx buffers,		\
+	      reduce CONFIG_BT_BUF_ACL_RX_COUNT_EXTRA or increase CONFIG_BT_BUF_CMD_TX_COUNT.");
+#else  /* controller-only build */
+// not BT_BUF_ACL_RX_COUNT, but what is set in controller.
+BUILD_ASSERT(CONFIG_BT_BUF_CMD_TX_COUNT > 2,
+	     "We need at least two HCI command buffers to avoid deadlocks.");
```

Comment:
```suggestion
BUILD_ASSERT(CONFIG_BT_BUF_CMD_TX_COUNT - 1 > 0,
	     "We need at least two HCI command buffers to avoid deadlocks.");
```
it's not about controller, right? It's that adjustment that should not become 0 https://github.com/nrfconnect/sdk-nrf/pull/19869/files#diff-813ccd03aac379c4e1f534c16d65b666f48162012fd09e35a1f1ddcf69d88511R22

Also I would still like to place both static assert and dynamic adjustment at the same place in the code. It's harder to find pieces and understated them when they scattered.